### PR TITLE
[MF] [KD] Render List Names and List Items

### LINKED ISSUE #2: 2. As a user, I want to read all of my shopping lists and the items in them.
## Summary

The most important feature of our app is the ability to look at their shopping lists to see what's on them, so the user knows what they need to buy. For this to work, we'll have to be able to choose a our shopping lists. The `List` (the active shopping list) and `Home` (where the user will be able to switch shopping lists) components are the best place in our code to do that. In fact, this feature is already partially implemented. Refer to the acceptance criteria in this issue, and the existing code, to understand the work that remains.

The `List` component is rendered on the `/list` page. Because the navigation links are currently broken, you'll have to type that path into your browser's address bar.

## Acceptance criteria
- [x] The `useShoppingLists` hook is used in `App.jsx` to get the shopping lists a user has access to from the database
- [x] The lists are passed into the `Home` component as a prop named `data`
- [x] The `setListPath` function is passed into the `Home` component so we can update the current list
- [x] The `useShoppingListData` hook is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [x] In `Home.jsx`, the `SingleList` component is used to render the name of each shopping list a user is subscribed to
- [x] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Notes
- ⚠️ Your **only goal** is to make sure that the names of the items render as expected. You might be tempted to modify `ListItem.jsx` or `SingleList.jsx` to show more information – you will do that **later**.
- The variables you need are already being passed into the `List` and `Home` components. What steps do you need to take in order to render them?
- Additionally, until issues [4](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/issues/4) and [5](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/issues/5) you should only be able to see the shopping list that was manually added in [issue 1](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/issues/1).

### PR DESCRIPTION
**Pull request by piecanoe**
## Description

This code renders List Names on Home and List Items on /list.

## Related Issue

Closes #2

## Acceptance Criteria

- [x] The `useShoppingLists` hook is used in `App.jsx` to get the shopping lists a user has access to from the database
- [x] The lists are passed into the `Home` component as a prop named `data`
- [x] The `setListPath` function is passed into the `Home` component so we can update the current list
- [x] The `useShoppingListData` hook is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [x] In `Home.jsx`, the `SingleList` component is used to render the name of each shopping list a user is subscribed to
- [x] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

### Before
Screenshot 2024-02-05 at 5 13 03 PM

Screenshot 2024-02-05 at 5 12 33 PM

### After
tcl69_issue#2_lists_fixed

tcl69_issue#2_listsitems_fixed

tcl69_issue#2_lists

tcl69_issue#2_listsitems

## Testing Steps / QA Criteria

After sign-in, conditional render will show List Names on Home or "You have no Lists!". On /list route, conditional render will show List Items or "You have no items on your list!".

### CODE REVIEW

File: src/views/Home.jsx [hunk 1]
```diff
@@ -7,10 +8,17 @@ export function Home({ data, setListPath }) {
 				Hello from the home (<code>/</code>) page!
 			</p>
 			<ul>
-				{/**
-				 * TODO: write some JavaScript that renders the `lists` array
-				 * so we can see which lists the user has access to.
-				 */}
+				{data ? (
```
Comment (amalyam) [inline src/views/Home.jsx]: Seems like this ternary is reversed

File: src/views/List.jsx [hunk 2]
```diff
@@ -7,11 +7,13 @@ export function List({ data }) {
...
-				 * using the `ListItem` component that's imported at the top
-				 * of this file.
-				 */}
+				{data ? (
+					<h1>You have no items in your list!</h1>
+				) : (
+					data.map((item, index) => (
+						<ListItem key={index} name={item.itemName} />
+					))
+				)}
```
Comment (amalyam) [inline src/views/List.jsx:16]: This ternary also seems to be reversed

**Review by amalyam — CHANGES_REQUESTED** [review]
The two ternary statements both seem to be in reverse order

File: src/views/Home.jsx (hunk 1 above)
Comment (krsnamara) [reply src/views/Home.jsx]: Yes! It is. Will update that

**Comment by hsiangj** [comment]
In the 'Notes' section in issue#2, it says there's a manual shopping list that could be used for rendering from issue#1. Perhaps this could be incorporated in List and List Items?

**Comment by krsnamara** [comment]
We noticed that the list items appeared to be saved in local storage or cached somewhere. Upon log out the, navigation to /list will still show the list items. Assumption is there needs to be an "onSignOut" destroy or delete function added somewhere.

File: src/views/Home.jsx [hunk 3]
```diff
@@ -2,22 +2,24 @@ import './Home.css';
...
 		<div className="Home">
 			<p>
 				Hello from the home (<code>/</code>) page!
 			</p>
 			<ul>
-				{data ? (
-					<h1>You have no Lists!</h1>
-				) : (
-					data.map((list, index) => (
+				{data && data.length > 0 ? (
```
Comment (amalyam) [inline src/views/Home.jsx:12]: Nice double check here

**Review by krsnamara — APPROVED** [review]

File: src/views/List.jsx (hunk 2 above)
Comment (amalyam) [reply src/views/List.jsx:16]: Fix looks good, and also additional mapping!

**Review by amalyam — APPROVED** [review]
Fix on ternaries looks good!

**Review by mxmason — APPROVED** [review]
Works as expected, and great job on the fix!

**Review by louieknolle — APPROVED** [review]
Works as expected!

**Pull request merged**